t-update-filter-behavior.html

```diff
@@ -26,61 +26,52 @@
                 } else {
-                    var context = inventories[i];
+                    context = inventory;
                 }
+
                 if (mappedData.executeFunctionName) {
-                    var totalFareComponent = this._executeFunctionByName(mappedData.executeFunctionName, this, context, mappedData, buildFilters);
+                    totalFareComponent = this._executeFunctionByName(mappedData.executeFunctionName, this, context, mappedData, buildFilters);
                 }
             }
-        }
-
+        });
 
         //update Filters existing filters
-        for (var i = 0; i < filters.length; i++) {
-            for(var filterCategory in buildFilters){
-                if (filterCategory == filters[i].category) {
-                    if (filters[i].type == "Range") {
-                        filters[i].items[0].value = buildFilters[filterCategory].min.value
-                        filters[i].items[0].displayValue = buildFilters[filterCategory].min.displayValue
-                        filters[i].items[1].value = buildFilters[filterCategory].max.value
-                        filters[i].items[1].displayValue = buildFilters[filterCategory].max.displayValue
-                    } else if(filters[i].type == "Options"){
-                        //Make count of options filter zero
-                        for (var j = 0; j < filters[i].items.length; j++) {
-                            if (filters[i].items[j].additionalInfo && filters[i].items[j].additionalInfo.length) {
-                                filters[i].items[j].additionalInfo[0].value = "0";
-                            }
-                        }
-                        
-                        //Add exact count
-                        for (var j = 0; j < filters[i].items.length; j++) {
-                            for(var filterLebel in buildFilters[filterCategory]){
-                                if (filters[i].items[j].label.toLowerCase() == filterLebel.toLowerCase()) {
-                                    if (filters[i].items[j].additionalInfo && filters[i].items[j].additionalInfo.length) {
-                                        filters[i].items[j].additionalInfo[0].value = buildFilters[filterCategory][filterLebel];
-                                    }
-                                }
-                            }
-                        }
+        filters.forEach(function(filter) {
+            var buildFilter = buildFilters[filter.category];
+
+            if (buildFilter && filter.type === "Range") {
+                filter.items[0].value = buildFilter.min.value
+                filter.items[0].displayValue = buildFilter.min.displayValue
+                filter.items[1].value = buildFilter.max.value
+                filter.items[1].displayValue = buildFilter.max.displayValue
+            }
+            else if (buildFilter && filter.type === "Options") {
+                filter.items.forEach(function(item) {
```

Comment:
We can get rid of nested for loop and two similar for loops by directly accessing the property and merging the bodies of both for loops.